p2p/server.go

```diff
@@ -102,13 +122,20 @@ func (srv *Server) Start() (err error) {
 	srv.addpeer = make(chan *Peer)
 	srv.delpeer = make(chan *Peer)
 
-	addr, err := net.ResolveUDPAddr("udp4", fmt.Sprintf("0.0.0.0:%s", srv.KadPort))
+	srv.PrivateKey, err = crypto.LoadECDSAFromString(srv.ECDSAKey)
 	if err != nil {
 		return err
 	}
 
-	myId := common.HexToAddress(srv.MyNodeID)
-	srv.kadDB = discovery.StartService(myId, addr, srv.StaticNodes)
+	buff := crypto.FromECDSAPub(&srv.PrivateKey.PublicKey)
+	srv.MyNodeID = "0x" + hex.EncodeToString(buff[1:])
```

Comment:
Recommend to define a Hex/HexString type handle handle all hex related operations.
In this case, we can avoid "0x" in the code everywhere.

Reply:
Create function to convert public key from binary to string format